### ISSUE TITLE
RxState should be readonly?

### ISSUE THREAD

**Issue by jon9090** [rx-angular/rx-angular#1449]
# Description

When I change the company object, it changes the store, also.

Which causes unexpected behavior in the state.

Why does the store not readonly? (using object freeze like ngrx)

```
vm$ = this.state.select(selectSlice(["company"])).pipe(
    map(({ company }) => ({ company })),
    tap(({ company }) => {
      company.id = 2; // <-------------- I change the id here, and it change the store because it's ref object.
      console.log({ company });
    })
  );
```

https://codesandbox.io/s/condescending-thompson-e51sh1?file=/src/app/app.component.ts:334-670

```
   template: `
    in app! vm = {{ vm$ | async | json }}<br />
    state = {{ state$ | async | json }}                          <---- state changed.
  `,
  providers: [RxState]
})
export class AppComponent {
  title = "CodeSandbox";

  state$ = this.state.select();

  vm$ = this.state.select(selectSlice(["company"])).pipe(
    map(({ company }) => ({ company })),
    tap(({ company }) => {
      company.id = 2; // <-- should be readonly??
      console.log({ company });
    })
  );

  constructor(private state: RxState<{ company: { id: number } }>) {
    this.state.set({ company: { id: 1 } });
  }
```

**Comment by hoebbelsB** [comment]
hey @jon9090 that's a very nice finding, thanks! Could it be solved by just typing it as readonly object? :thinking:

**Comment by jon9090** [comment]
@hoebbelsB thanks :)
I think it won't be enough because I select the state then I pass it to `FormGroup` and inside the controls can change the value..
If you want I can provide codesandbox example.

**Comment by vmasek** [comment]
I like the idea of typing the state as `readonly` but I don't really understand the need to force (lock) the object on runtime. It actually comes with performance downside so I would not consider it a viable feature request.

**Comment by edbzn** [comment]
+1 for `readonly` typing but not in runtime

**Comment by edbzn** [comment]
Fixed by #1618, thanks for raising this.
